test/unit/ngsiv2/plugins/alias-plugin_test.js

```diff
@@ -144,6 +180,239 @@ describe('Attribute alias plugin', function() {
+        var values = [
+            {
+                name: 'ut',
+                type: 'Number',
+                value: '99823423'
+            }
+        ];
+
+        beforeEach(function() {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities/light1/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateContextAliasPlugin3.json'))
+                .reply(204);
+        });
+
+        it('should rename the attributes as expected by the mappings', function(done) {
+            iotAgentLib.update('light1', 'Light', '', values, function(error) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When an update comes for attributes with aliases and integer type.' + 
+        'The IOTA sends the update with alias and using JSON native type', function() {
+        var values = [
+            {
+                name: 'ut',
+                type: 'Number',
+                value: '99823423'
+            }
+        ];
+
+        beforeEach(function() {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities/light1/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateContextAliasPlugin3.json'))
+                .reply(204);
+        });
+
+        it('should rename the attributes as expected by the mappings', function(done) {
+            iotAgentLib.update('light1', 'Light', '', values, function(error) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When an update comes for attributes with aliases and float type', function() {
+        var values = [
+            {
+                name: 'ap',
+                type: 'Float',
+                value: '0.45'
+            }
+        ];
+
+        beforeEach(function() {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities/light1/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateContextAliasPlugin4.json'))
+                .reply(204);
+        });
+
+        it('should rename the attributes as expected by the mappings', function(done) {
```

Comment:
"should rename the attribute"...

In fact, it is a matter of casting the value, but the name of the attribute doesn't change. "should cast the attribute value as expected..." seems to be more accurate.

Reply:
I think that the expression is correct because it does not refer to the casting but to the alias plugin functionality, i.e., we receive an observation using `ap` as name and the plugin renames it to `active_power`.

Reply:
I understand... Thus, maybe the precise wording is a matter of adding "an cast value" in the proper place, I mean:

"should rename the attributes as expected by the mappings" -> "should rename the attributes and cast value as expected by the mappings"

Anywaya, if still don't liking it, then just NTC and keep the current wording :)

Reply:
Fixed in 63aa7b0bf530c78fd0825f01f831e12992746b2f